[
  {
    "project": "machinalis/mypy-django",
    "commit": "cce9e2a85edb2ccce60c608676288ec8597e0f52",
    "filename": "django/core/files/uploadedfile.pyi",
    "filename_after_commit": "/home/chowyi/TypeAnnotation_Study/GitHub/machinalis-mypy-django/django/core/files/uploadedfile.pyi",
    "file_hunks_size": 1,
    "min_patch_found": false,
    "single_hunk": false,
    "fit_TFix": false,
    "delete_only_patch": false,
    "has_suppression_all_hunks": false,
    "full_warning_msg": "django/core/files/uploadedfile.pyi:11:23 Incompatible variable type [9]: file is declared to have type `IO[typing.Any]` but is used as type `None`.",
    "message": " file is declared to have type `IO[typing.Any]` but is used as type `None`.",
    "rule_id": "Incompatible variable type [9]",
    "warning_line_no": 11,
    "warning_line": "    def __init__(self, file: IO=None, name: str=None, content_type: str=None, size: int=None, charset: str=None, content_type_extra: Dict[str, str]=None) -> None: ..."
  }
]